lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1119,24 +1110,24 @@ def _generate_invocation_bco(self, trans, invocation_id, **kwd):
                                 }
                             }
                             output_subdomain.append(output)
-                current_tool = dict_workflow['steps'][str(i)]
+                current_step = dict_workflow['steps'][str(i)]
                 pipeline_step = {
                     'step_number': i,
-                    'name': current_tool['name'],
-                    'description': current_tool['annotation'],
-                    'version': current_tool['tool_version'],
+                    'name': current_step['name'],
+                    'description': current_step['annotation'],
+                    'version': current_step['tool_version'],
```

Comment:
I don't think steps have a tool_version, maybe skip this ? That should be what causes https://jenkins.galaxyproject.org/job/docker-api-py3/5470/testReport/junit/lib.galaxy_test.api.test_workflows/WorkflowsApiTestCase/test_run_subworkflow_simple/

Reply:
From the little that I have seen the following line: https://github.com/galaxyproject/galaxy/blob/2f2923d4043dc305a664cb80c91d0a74e3dbec66/lib/galaxy/webapps/galaxy/api/workflows.py#L1077
caused this to work in the limited tests I was able to run. Even pulling in a `None` some times, but I guess that was because I was always using tools from the Tool Shed. For reproducibility the version is important. Do you think we could  find some workaround or alternate solution as opposed to skipping it?

Reply:
My bad, tools may have steps in the exported workflow dict (I'd personally would have worked with the workflow, but that's fine for now), but not if the step is not a tool step. The order of the steps was not correct though, so when you pick out the workflow_dict step using the index (i) of the workflow step you may or may not get the step you think you're getting, so if you pick a non tool step this errors out. I've fixed that in the last commit.